contracts/SaleShare.sol

```diff
@@ -33,6 +34,7 @@ contract SaleShare is IERC677Receiver, Ownable, ReentrancyGuard {
     uint public totalScore;
     uint public totalAmount;
     uint public finalizedBaseAmount;
+    uint public finalizedTotalAmount;
```

Comment:
Why would we need to use a `finalizedTotalAmount` in the calculations for `owed`?

I'm probably missing something obvious here but if we are basing the allocation on the `totalAmount` in the calculation of the `finalizedBaseAmount` wouldn't numbers now be wrong if `getUserInfo` now uses a potentially different total amount?

Reply:
Nevermind, after staring at it some more I now see how, we can't use the `totalAmount` as it might include refunds that are not to counted as allocation amount.

Now it seems like could use `raisingAmount` instead no? Except for when the sale doesn't sell out, in which case allocations should scale based on the `totalAmount` reached. So instead of `finalizedTotalAmount` could we use `_min(raisingAmount, totalAmount)`?

Or are we simply trying to be precise and account for the fact that incrementing the base amount by 0.1 will never give us exactly `raisingAmount` in the end?